base/applications/rapps/applicationinfo.cpp

```diff
@@ -0,0 +1,626 @@
+        GetApplicationRegString(L"DisplayVersion", szDisplayVersion);
+        GetApplicationRegString(L"Comments", szComments);
+    }
+}
+
+CInstalledApplicationInfo::~CInstalledApplicationInfo()
+{
+}
+
+VOID
+CInstalledApplicationInfo::AddApplicationRegString(CAppRichEdit *RichEdit, UINT StringID, const CStringW &String, DWORD TextFlags)
+{
+    CStringW Tmp;
+    if (GetApplicationRegString(String, Tmp))
+    {
+        RichEdit->InsertTextWithString(StringID, Tmp, TextFlags);
+    }
+}
+
+
+VOID
+CInstalledApplicationInfo::ShowAppInfo(CAppRichEdit *RichEdit)
+{
+    RichEdit->SetText(szDisplayName, CFE_BOLD);
+    RichEdit->InsertText(L"\n", 0);
+
+    RichEdit->InsertTextWithString(IDS_INFO_VERSION, szDisplayVersion, 0);
+    AddApplicationRegString(RichEdit, IDS_INFO_PUBLISHER, L"Publisher", 0);
+    AddApplicationRegString(RichEdit, IDS_INFO_REGOWNER, L"RegOwner", 0);
+    AddApplicationRegString(RichEdit, IDS_INFO_PRODUCTID, L"ProductID", 0);
+    AddApplicationRegString(RichEdit, IDS_INFO_HELPLINK, L"HelpLink", CFM_LINK);
+    AddApplicationRegString(RichEdit, IDS_INFO_HELPPHONE, L"HelpTelephone", 0);
+    AddApplicationRegString(RichEdit, IDS_INFO_README, L"Readme", 0);
+    AddApplicationRegString(RichEdit, IDS_INFO_CONTACT, L"Contact", 0);
+    AddApplicationRegString(RichEdit, IDS_INFO_UPDATEINFO, L"URLUpdateInfo", CFM_LINK);
+    AddApplicationRegString(RichEdit, IDS_INFO_INFOABOUT, L"URLInfoAbout", CFM_LINK);
+    RichEdit->InsertTextWithString(IDS_INFO_COMMENTS, szComments, 0);
+
+    if (m_szInstallDate.IsEmpty())
+    {
+        RetrieveInstallDate();
+    }
+
+    RichEdit->InsertTextWithString(IDS_INFO_INSTALLDATE, m_szInstallDate, 0);
+    AddApplicationRegString(RichEdit, IDS_INFO_INSTLOCATION, L"InstallLocation", 0);
+    AddApplicationRegString(RichEdit, IDS_INFO_INSTALLSRC, L"InstallSource", 0);
+
+    if (m_szUninstallString.IsEmpty())
+    {
+        RetrieveUninstallStrings();
+    }
+
+    RichEdit->InsertTextWithString(IDS_INFO_UNINSTALLSTR, m_szUninstallString, 0);
+    RichEdit->InsertTextWithString(IDS_INFO_MODIFYPATH, m_szModifyString, 0);
+}
+
+VOID
+CInstalledApplicationInfo::RetrieveInstallDate()
+{
+    DWORD dwInstallTimeStamp;
+    SYSTEMTIME InstallLocalTime;
+    if (GetApplicationRegString(L"InstallDate", m_szInstallDate))
+    {
+        ZeroMemory(&InstallLocalTime, sizeof(InstallLocalTime));
+        // Check if we have 8 characters to parse the datetime.
+        // Maybe other formats exist as well?
+        m_szInstallDate = m_szInstallDate.Trim();
+        if (m_szInstallDate.GetLength() == 8)
+        {
+            InstallLocalTime.wYear = wcstol(m_szInstallDate.Left(4).GetString(), NULL, 10);
+            InstallLocalTime.wMonth = wcstol(m_szInstallDate.Mid(4, 2).GetString(), NULL, 10);
+            InstallLocalTime.wDay = wcstol(m_szInstallDate.Mid(6, 2).GetString(), NULL, 10);
+        }
+    }
+    // It might be a DWORD (Unix timestamp). try again.
```

Comment:
The comment is very good regarding its content, but its positioning (before the else) is awful. Please move it!